examples/gnrc_tcp_cli/main.c

```diff
@@ -0,0 +1,235 @@
+                      (void *)i, NULL);
+    }
+
+    return 0;
+}
+
+void *cli_thread(void *arg)
+{
+    /* Test Program variables */
+    int32_t tid = (int) arg;
+    uint32_t rcvd = 0;
+    uint32_t sent = 0;
+    uint32_t cycles = 0;
+    uint32_t cycles_ok = 0;
+
+    /* TCP specific variables */
+    ipv6_addr_t local_addr;
+    ipv6_addr_t peer_addr;
+    gnrc_tcp_tcb_t tcb;
+
+    /* Target Peer Address Information */
+    ipv6_addr_t target_addr;
+    uint16_t target_port;
+
+    /* Initialize Target Information */
+    ipv6_addr_from_str(&target_addr, TARGET_ADDR);
+    target_port = TARGET_PORT;
+
+    printf("client thread running: TID=%"PRIi32"\n", tid);
+    while (cycles < CYCLES) {
+        /* Initialize tcb struct */
+        int32_t ret = gnrc_tcp_tcb_init(&tcb, AF_INET6, (uint8_t *) &local_addr, (uint8_t *) &peer_addr);
+        switch (ret) {
+            case 0:
+                DEBUG("TID=%"PRIi32" : gnrc_tcp_tcb_init() : 0 : ok\n", tid);
+                break;
+
+            case -EAFNOSUPPORT:
+                printf("TID=%"PRIi32" : gnrc_tcp_tcb_init() : -EAFNOSUPPORT\n", tid);
+                continue;
+
+            default:
+                printf("TID=%"PRIi32" : gnrc_tcp_tcb_init() : %"PRIi32"\n", tid, ret);
+                continue;
+        }
+
+        /* Connect to Peer */
+        ret = gnrc_tcp_open_active(&tcb, AF_INET6, (uint8_t*) &target_addr, target_port, 0);
+        switch (ret) {
+            case 0:
+                DEBUG("TID=%"PRIi32" : gnrc_tcp_open_active() : 0 : ok\n", tid);
+                break;
+
+            case -EISCONN:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EISCONN\n", tid);
+                continue;
+
+            case -EINVAL:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EINVAL\n", tid);
+                continue;
+
+            case -EAFNOSUPPORT:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EAFNOSUPPORT\n", tid);
+                continue;
+
+            case -EADDRINUSE:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -EADDRINUSE\n", tid);
+                continue;
+
+            case -ECONNREFUSED:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -ECONNREFUSED : retry 5sec\n", tid);
+                xtimer_sleep(5);
+                continue;
+
+            case -ENOMEM:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -ENOMEM\n", tid);
+                continue;
+
+            case -ETIMEDOUT:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : -ETIMEDOUT\n", tid);
+                continue;
+
+            default:
+                printf("TID=%"PRIi32" : gnrc_tcp_open_active() : %"PRIi32"\n", tid, ret);
+                continue;
+        }
+
+        /* Fill Buffer with a test pattern */
+        for (uint32_t i=0; i < sizeof(bufs[tid]) - 1; i+= 2){
```

Comment:
I think `int i = 0` should suffice, no need for explicit size of 4 bytes here, or do you expect to handle 4 billion connections?